### ISSUE TITLE
androidTest fails in new Android Studio Chipmunk release 2021.2.1

### ISSUE THREAD

**Issue by MikeMcC399** [corona-warn-app/cwa-app-android#5181]
## Technical details

cwa-android branch: [release/2.23.x](https://github.com/corona-warn-app/cwa-app-android/tree/release/2.23.x)
Android Studio Chipmunk | 2021.2.1
Windows 11

## Describe the bug

Using the Android Studio Chipmunk release 2021.2.1, classed as "stable" and released on May 9, 2022 (see https://android-developers.googleblog.com/2022/05/android-studio-chipmunk.html) CWA Android [release/2.23.x](https://github.com/corona-warn-app/cwa-app-android/tree/release/2.23.x) produces the error message `Duplicate content roots detected`. Attempting to run `androidTests` produces the error message `SEVERE: Test execution failed with fatal error!` and it is not possible to run the tests.

This is a regression from the previous stable environment Android Studio Bumblebee | 2021.1.1 Patch 3. It affects build variants
- `deviceDebug`
- `deviceForTestersDebug`

(`deviceRelease` and `deviceForTestersRelease` do not include `androidTest`.)

### Android Studio Chipmunk

Build > Select Build Variant: `deviceforTestersDebug`

File > Sync Project with Gradle Files

Note error message:
```
Duplicate content roots detected: Path [C:/Users/<user_name>/github/cwa-app-android/Corona-Warn-App/src/testShared/java] of module [Corona-Warn.Corona-Warn-App.unitTest] was removed from modules [Corona-Warn.Corona-Warn-App.androidTest]
```

Build > Rebuild Project
Run androidTest (Tests in `de/rki/coronawarnapp`) on Android 11 device

Note error message:
```
com.google.testing.platform.RunnerImpl run
SEVERE: Test execution failed with fatal error!
com.google.testing.platform.api.plugin.PluginException: Exception thrown during onBeforeAll invocation of plugin com.google.testing.platform.plugin.android.AndroidDevicePlugin.
```

### Android Studio Bumblebee

For comparison, the identical steps are successful in Android Studio Bumblebee.

Build > Select Build Variant: `deviceforTestersDebug`
File > Sync Project with Gradle Files<br>
(no error message)<br>
Build > Rebuild Project<br>
Run androidTest (Tests in de/rki/coronawarnapp) on Android 11 device

Status shows 7 failed, 181 passed, 253 tests

## Expected behavior

It should be possible to run `androidTest` in the stable release of Android Studio, currently the Chipmunk 2021.2.1 version.

## Possible Fix

There is a related P1 bug logged on https://issuetracker.google.com/issues/232007221 "Duplicate content roots detected" with Android Gradle plugin 7.2.0", however CWA is using AGP 7.1.3 not 7.2.0.

If this is not fixable in CWA Android, then a note should be added to https://github.com/corona-warn-app/cwa-app-android/wiki/1-Setup which recommends using the latest version of Android Studio. That recommendation should be changed.

---
Internal Tracking ID: [EXPOSUREAPP-13104](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-13104)

**Comment by MikeMcC399** [comment]
Does the comment from:

https://issuetracker.google.com/issues/232007221#comment6
"WRT sharing the same folder between unit tests and device tests, please note that is not supported in the IDE. A single source directory can belong to only 1 IDE module, ..."

apply to the following?

https://github.com/corona-warn-app/cwa-app-android/blob/802b59342afd968dd4605d26c5ac76fee51ea6f2/Corona-Warn-App/build.gradle#L258-L265

**Comment by dsarkar** [comment]
Hi @MikeMcC399, thanks for reporting, your issues has been forwarded Internal Tracking ID: [EXPOSUREAPP-13104](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-13104). Best, DS

**Comment by mtwalli** [comment]
@MikeMcC399  Thanks for reporting this issue, We are aware of this issue , however we can not upgrade to the latest AGP due to another issue that was reported to Google already see https://issuetracker.google.com/issues/232107695
in the mean time if you would like to run the `androidTest` use previous version of AGP (our current version)

**Comment by thorbenkuro** [comment]
@MikeMcC399 
Thank you for reporting the issue! I would suggest to change the download recommendation of the wiki from
> Download the latest version from Android Studio here: https://developer.android.com/studio.

to

> Download Android Studio Bumblebee (2021.1.1) Patch 3 from the download archives: https://developer.android.com/studio/archive  We recommend you to use this version, because we have reports about incompatibility problems with the latest release of Android Studio.

to make sure, that anyone can build the app without running into any errors.

As a side note: I tried to reproduce the error with Android Studio Chipmunk (2021.2.1), [release/2.23.x](https://github.com/corona-warn-app/cwa-app-android/tree/release/2.23.x) branch and build variant `deviceForTestersDebug` on Mac (using the default AGP 7.1.3) and for me `androidTest` did produce an error at first, that looks like the one you mentioned. Looking at the stack trace it seems like the emulator (Pixel 4a, Android 12) did not have enough free space available. I wiped the data of the emulator device to fix this and everything seems to be back to normal (255 Tests, 248 passed, 7 failed) on Android Studio Chipmunk for me.

**Comment by MikeMcC399** [comment]
I am closing this issue, since the fatal error is not always reproducible.

I found even with Bumblebee that the results of the running the complete `androidTest` were inconsistent from one run to the next run. Sometimes tests would fail in the complete test run, whereas if the same test was run individually it passed. 

As previously stated, the error `Duplicate content roots detected:` is always reproducible. It is not clear to me though if this is a harmful condition.

**Comment by MikeMcC399** [comment]
@mtwalli

See comment https://issuetracker.google.com/issues/232007221#comment18 from Google that the configuration in CWA Android which causes the error `Duplicate content roots detected:` is no longer supported.

**Comment by MikeMcC399** [comment]
- This issue is resolved in release 2.24 through PR #5238. The message  `Duplicate content roots detected:.` is no longer output in Android Studio Chipmunk, 2021.2.1 Patch 1 when performing a "Sync Project with Gradle Files" operation.

Closing again, since this is a development-only issue, which does not affect end-users.
